src/commands/test_balances.rs

```diff
@@ -1,58 +1,114 @@
+use crate::components::transactions::{create_burn_transaction, create_transfer_transaction};
 use crate::components::user::User;
 use crate::prelude::info;
 use orchard::keys::Scope::External;
 use orchard::note::AssetBase;
+use zcash_primitives::transaction::Transaction;
 
-#[derive(Debug, Copy, Clone)]
-pub(crate) struct TestBalances {
-    account0: i64,
-    account1: i64,
-}
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct TestBalances(Vec<i64>);
 
 impl TestBalances {
-    pub(crate) fn new(account0: i64, account1: i64) -> Self {
-        TestBalances { account0, account1 }
+    pub(crate) fn add_balances(&mut self, balances: Vec<(u32, i64)>) {
+        balances.iter().for_each(|(index, balance)| {
+            assert!((*index as usize) < self.0.len());
+            self.0[*index as usize] += *balance;
+        });
     }
 
-    pub(crate) fn get_zec(user: &mut User) -> TestBalances {
-        Self::get_asset(AssetBase::native(), user)
+    pub(crate) fn get_zec(user: &mut User, num_users: u32) -> TestBalances {
+        Self::get_asset(AssetBase::native(), user, num_users)
     }
 
-    pub(crate) fn get_asset(asset: AssetBase, wallet: &mut User) -> TestBalances {
-        let address0 = wallet.address_for_account(0, External);
-        let address1 = wallet.address_for_account(1, External);
+    pub(crate) fn get_asset(asset: AssetBase, wallet: &mut User, num_users: u32) -> TestBalances {
```

Comment:
Let's make 'user' either leftmost or rightmost parameter in both methods

Reply:
Moved to rightmost